transform.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * go-leia
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package leia
+
+import (
+	"regexp"
+	"strings"
+)
+
+// Transform is a function definition for transforming values and search terms.
+type Transform func(string) string
+
+// NoTransform is the default Transform function and returns the given data as is.
```

Comment:
You check for `nil` in the `Transform` func (goes for `Tokenize` as well) so you actually don't need this (both `NoTransform` or passing `nil` lead to the same behaviour)

Reply:
I started with a clean default func, later I still needed the nil check. Removed the default funcs and just rely on the nil check.